src/main/groovy/com/avioconsulting/mule/linter/rule/configuration/UnusedFlowRule.groovy

```diff
@@ -0,0 +1,97 @@
+    static final String APIKIT_FLOW_PREFIX_REGEX = '(get:|post:|put:|patch:|delete:|head:|options:|trace:).*'
+    UnusedFlowRule() {
+        this.ruleId = RULE_ID
+        this.ruleName = RULE_NAME
+    }
+
+    List<MuleComponent> getAllFlow(Application app) {
+        List<MuleComponent> allFlowsSubflows = []
+        app.configurationFiles.each { configFile ->
+            flowSubFlowComponent.each { element ->
+                List<MuleComponent> flowsSubflows = configFile.findComponents(element.key, element.value)
+                allFlowsSubflows += flowsSubflows
+            }
+        }
+        return allFlowsSubflows
+    }
+
+    List<MuleComponent> getAllFlowRef(Application app) {
+        List<MuleComponent> allFlowRefs = []
+        app.configurationFiles.each { configFile ->
+            List<MuleComponent> flowRefs = configFile.findComponents('flow-ref',
+                                                                            ConfigurationFile.MULE_CORE_NAMESPACE)
+            allFlowRefs += flowRefs
+        }
+        return allFlowRefs
+    }
+
+    List<MuleComponent> notUsedByFlowRef(List<MuleComponent> allFlowsSubflows, List<MuleComponent> allFlowRefs) {
+        List<MuleComponent> notUsedByFlowRef = []
+        allFlowsSubflows.each { flow->
+            String flowSubflowName = flow.getAttributeValue('name')
+            List<MuleComponent> foundRef = allFlowRefs.findAll { it.getAttributeValue('name') == flowSubflowName }
+
+            if ( foundRef.size <= 0 ) {
+                notUsedByFlowRef += flow
+            }
+        }
+        return notUsedByFlowRef
+    }
+
+    List<MuleComponent> filterWithoutSource(List<MuleComponent> muleComponent) {
+        List<MuleComponent> withoutSourceComp = []
+        muleComponent.each { comp->
+            MuleComponent firstComp = comp.children[0]
+            if ( !firstComp.componentName.toLowerCase().contains('listener') ) {
+                withoutSourceComp += comp
+            }
+        }
+        return withoutSourceComp
+    }
+
+    List<MuleComponent> filterApiKitRouterFlow(List<MuleComponent> muleComponent) {
+        List<MuleComponent> withoutApiKitRouterFlow = []
+        muleComponent.each { comp->
+            if ( !comp.getAttributeValue('name').toLowerCase().matches(APIKIT_FLOW_PREFIX_REGEX)) {
+                withoutApiKitRouterFlow += comp
+            }
+        }
+        return withoutApiKitRouterFlow
+    }
+
+    @Override
+    List<RuleViolation> execute(Application app) {
+        List<RuleViolation> violations = []
+
+        List<MuleComponent> allFlowsSubflows = getAllFlow(app)
+
+        List<MuleComponent> allFlowRefs = getAllFlowRef(app)
+
+        List<MuleComponent> notUsedByFlowRef = notUsedByFlowRef(allFlowsSubflows, allFlowRefs)
+
+        List<MuleComponent> withoutSourceComp = filterWithoutSource(notUsedByFlowRef)
+
+        List<MuleComponent> withoutApiKitRouterFlow = filterApiKitRouterFlow(withoutSourceComp)
+
```

Comment:
It seems a little overkill to create 5 different lists... I understand if this was for debugging purpose... but you should only need 2... 1 for flow/sub-flows that aren't apikit and don't have a listener... you should be able to filter these out in the getAllFlow method (although might want to rename it)...

Then you should be able to just do something like:
List diffs = flows - allFlowRefs